File: CD/projeto/APIWeb/gis/src/main/java/pt/isel/ps/gis/hypermedia/siren/components/subentities/Action.java
package pt.isel.ps.gis.hypermedia.siren.components.subentities;

import com.fasterxml.jackson.annotation.JsonInclude;

@JsonInclude(JsonInclude.Include.NON_NULL)
public class Action {

    private String name;
    private String title;
    private Method method;
    private String href;
    private String type;
    private Field[] fields;

    public Action(String name, String title, Method method, String href, String type, Field[] fields) {
        this.name = name;
        this.title = title;
        this.method = method;
        this.href = href;
        this.type = type;
        this.fields = fields;
    }

    public String getName() {
        return name;
    }

    public String getTitle() {
        return title;
    }

    public Method getMethod() {
        return method;
    }

    public String getHref() {
        return href;
    }

    public String getType() {
        return type;
    }

    public Field[] getFields() {
        return fields;
    }
}

File: CD/projeto/AppsAndroid/gis/app/src/main/java/ps/leic/isel/pt/gis/hypermedia/siren/subentities/Action.kt
package ps.leic.isel.pt.gis.hypermedia.siren.subentities;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties

@JsonIgnoreProperties(ignoreUnknown = true)
class Action(val name: String, val title: String?, val method: Method?, val href: String,
             val type: String?, val fields: Array<Field>?)
